.github/workflows/milestone.yml

```
@@ -34,22 +34,31 @@ jobs:
                 commit_sha: context.sha
             })
             if (pr_response.data.length === 0) {
-                console.log('Pull request not found')
+                console.log('Pull request not found for commit', context.sha)
                 return
             }
             if (pr_response.data.length > 1) {
                 console.log(pr_response.data)
-                throw 'Expect 1 pull request but found: ' + pr_response.data.length
+                throw 'Expected 1 pull request but found: ' + pr_response.data.length
             }
+            const pr_number = pr_response.data[0].number
+            console.log('Found pull request', pr_number, 'for commit', context.sha)
 
             // Get milestone
             const {
                 MILESTONE_NUMBER
             } = process.env
+            console.log('Milestone number to set', MILESTONE_NUMBER)
 
-            // Find milestone
-            const response = await github.rest.issues.listMilestones(context.repo)
-            let milestone = response.data.find(milestoneResponse => milestoneResponse.title === MILESTONE_NUMBER)
+            // Find milestone with matching title
+            // See https://octokit.github.io/rest.js/v19#pagination
+            const milestones = await github.paginate(github.rest.issues.listMilestones, {
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                state: 'all'
```

How about also sorting ascending by completeness, for a higher chance the open ones will be on page 1?
```suggestion
                state: 'all',
                sort: 'completeness',
                direction: 'asc'
```
Ref: https://octokit.github.io/rest.js/v19#issues-list-milestones

---

There's no benefit since the iterator is `paginate` pulls all pages.

There's a mechanism to stop iterating early but I couldn't get it to work. See "to stop paginating" early on https://octokit.github.io/rest.js/v19#pagination.